src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -0,0 +1,257 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    internal class MsQuicApi : IDisposable
+    {
+        private bool _disposed;
+
+        private IntPtr _registrationContext;
+
+        internal unsafe MsQuicApi()
+        {
+            uint status = (uint)MsQuicNativeMethods.MsQuicOpen(version: 1, out MsQuicNativeMethods.NativeApi* registration);
+            MsQuicStatusException.ThrowIfFailed(status);
+
+            NativeRegistration = *registration;
```

Comment:
Do you really need to store a copy of `NativeRegistration` into a field? I do not see it used anywhere.